nixos/modules/programs/rust-motd.nix

```diff
@@ -28,9 +55,35 @@ in {
       '';
     };
     settings = mkOption {
-      type = types.submodule {
+      type = types.attrsOf (types.submodule {
         freeformType = format.type;
-      };
+        options.priority = mkOption {
+          type = types.int;
+          default = modules.defaultOrderPriority;
+          description = mdDoc ''
+            In `rust-motd`, the order of the sections in TOML correlates to the order
+            of the items displayed in the resulting `motd`. Attributes in Nix are
+            ordered alphabetically, e.g. `banner` would always be before `uptime`.
```

Comment:
Oof, I looked into this, apparently it's not really specified in toml whether the order should be preserved. This here from [the table spec](https://toml.io/en/v1.0.0#table) is the only mention regarding the order, but it's not exactly that:

> Key/value pairs within tables are not guaranteed to be in any specific order.

In any case, the `toml` crate, which is used by rust-motd doesn't preserve the order by default, _but_ you [can enable it](https://docs.rs/toml/latest/toml/type.Table.html) to be preserved with a feature flag, which rust-motd indeed enables.

The origin of this flag can be tracked back to [this issue](https://github.com/toml-rs/toml-rs/issues/232) requesting the feature. In the discussion one of the use cases (and indeed one that makes sense) is to preserve the ordering when programmatically editing toml files.

However, for the purpose of being able to change behavior based on the order, that's just not great imo and should never have been allowed, because it leads to configurations that can't be nicely composed and is most likely not what the toml spec intended. This is really ugly here now because of that, when there's better ways of defining orderings, the best of which is probably just a list of elements like

```toml
order = [
  "banner",
  "uptime",
]
```

Anyways, it would be great to report this to the toml spec, rust-motd, the toml crate, etc., asking them to consider not relying on the order of keys for determining behavior. 

---

And as for a Nix-specific workaround, I think having such an `order` option of sorts would give a cleaner interface:
```nix
programs.rust-motd = {
  settings.banner = ...;
  settings.uptime = ...;
  order = [
    "banner"
    "uptime"
  ];
}
```

Reply:
Fair, that's indeed clearer, updated the code accordingly.

Reply:
> Key/value pairs within tables are not guaranteed to be in any specific order.

Hmm, reading the standard now, this seems to refer to the keys inside a table, i.e.

```toml
[foo]
bar = 1
baz = 2
```

where `bar`/`baz` are not guaranteed to be in a specific order.

Reply:
Oh and in this case it's the order of tables themselves, yeah I didn't see anything about that in the specification. I'd still use the same reasoning though, the order of tables shouldn't have an influence.